### PR TITLE
Edge Modifier Bug Fix

### PR DESCRIPTION
**Pull request by aalexandrasilva**
The following features have been changed:
1) When an edge removal or addition fails, it is now returning null and assigning the error score a maximum value
2) The graph is now also being saved before the optimization process

### CODE REVIEW

File: src/main/java/org/aksw/simba/lemming/ColouredGraphWrapper.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import java.util.List;
+
+import grph.Grph;
+
+/**
+ * Wrapper Class to enable the persistence of the ColouredGraph class
+ * @author Ana
+ *
+ */
+public class ColouredGraphWrapper implements Serializable {
```
Comment (MichaelRoeder) [inline src/main/java/org/aksw/simba/lemming/ColouredGraphWrapper.java:19]: I do not understand why this class is necessary. Wouldn't it be sufficient to make the ColouredGraph class implement the Serializable interface? :thinking:

File: src/main/java/org/aksw/simba/lemming/colour/ColourPaletteWrapper.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+ *
+ */
+public class ColourPaletteWrapper implements Serializable{
+	
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = 3744241658289486040L;
+	protected Map<String, BitSet> uriColourMap = new HashMap<String, BitSet>();
+    protected int highestColourId = 0;
```
Comment (MichaelRoeder) [inline src/main/java/org/aksw/simba/lemming/colour/ColourPaletteWrapper.java:20]: Do we really need to keep track of this id? Isn't it automatically define by the size of the uriColourMap?

File: src/main/java/org/aksw/simba/lemming/colour/ColourPaletteWrapper.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import java.util.Map;
+import java.io.Serializable;
+import java.util.BitSet;
+
+/**
+ * Wrapper Class to enable the persistence of the ColourPalette class
+ * @author Ana
+ *
+ */
+public class ColourPaletteWrapper implements Serializable{
```
Comment (MichaelRoeder) [inline src/main/java/org/aksw/simba/lemming/colour/ColourPaletteWrapper.java:13]: Again: do we really need the wrapper class?

**Review by MichaelRoeder — COMMENTED** [review]
It looks good. I only have some small questions.

File: src/main/java/org/aksw/simba/lemming/ColouredGraphWrapper.java (hunk 1 above)
Comment (aalexandrasilva) [reply src/main/java/org/aksw/simba/lemming/ColouredGraphWrapper.java:19]: At the time, I found several usages of BitSet, ObjectObjectOpenHashMap, ObjectArrayList from com.carrotsearch.hppc which do not implement Serializable. I created these Wrapper classes to use java.util classes, analogous to the hppc ones, thus allowing serialization. :)

**Pull request merged**